### PR TITLE
Feature (Helm): Add library chart

### LINKED ISSUE #5085: Feature: Add library charts
The gist of this is in [the v3 proposal](https://github.com/helm/community/blob/master/helm-v3/001-charts.md#library-charts)

Note, the proposal refers to `requirements.yaml` files. Those files have now been merged into `Chart.yaml` files.

The gist is that library charts, that follow the idea of the [common chart](https://github.com/helm/charts/tree/master/incubator/common), can be used to share and re-use functionality. Library charts are not installable.

### PR DESCRIPTION
**Pull request by hickeyma**
**What this PR does / why we need it**:
This PR differentiate between an application chart and a library chart by defining a new `type` field in `Chart.yaml` . Library charts can be used as dependencies but are not installable.

Closes  #5085
partial #5397

*Special notes for your reviewer**:
Update on design proposal: https://github.com/helm/helm/issues/5085#issuecomment-462385310

*If applicable**:
- [x] this PR contains documentation
- [x] this PR contains unit tests
- [x] this PR has been tested for backwards compatibility

### CODE REVIEW

File: cmd/helm/install.go [hunk 1]
```diff
@@ -231,6 +231,14 @@ func (o *installOptions) run(out io.Writer) error {
 		return err
 	}
 
+	chartType := chartRequested.Metadata.Type
```
Comment (technosophos) [inline cmd/helm/install.go]: I see these same tests multiple times in this PR. Can we just make a function?

**Review by technosophos — APPROVED** [review]
I'm good with this. I left a minor nit comment, but all is looking good for adding this as a type.

File: cmd/helm/install.go (hunk 1 above)
Comment (hickeyma) [reply cmd/helm/install.go]: Good idea. Function added.

**Comment by hickeyma** [comment]
Thanks @technosophos for the review and comment. 

Review comment updated and ready for review again.

**Review by adamreese — APPROVED** [review]

**Comment by hickeyma** [comment]
Thanks @adamreese for merging.

**Pull request merged**